web/websocket.ex

```diff
@@ -21,26 +21,56 @@ defmodule Testgear.Websocket do
   @impl true
   def init(%Conn{request: request, context: context}) do
     Logger.info("should be able to emit log in websocket connection process")
-    :ok = Unique.register(request.query_params["name"], context)
+    if Map.has_key?(request.query_params, "name") do
+      :ok = Unique.register(request.query_params["name"], context)
+    else
+      :ok = Group.join(request.query_params["group_name"], context)
+    end
```

Comment:
(optional)
How about using pattern-matching?

```suggestion
    :ok = case request.query_params do
      %{"name" => name}             -> Unique.register(name, context)
      %{"group_name" => group_name} -> Group.join(group_name, context)
    end
```